core/app/models/spree/product.rb

```diff
@@ -213,8 +216,7 @@ def set_property(property_name, property_value, property_presentation = property
     end
 
     def possible_promotions
-      promotion_ids = promotion_rules.map(&:promotion_id).uniq
-      Spree::Promotion.advertised.where(id: promotion_ids).reject(&:expired?)
+      advertised_promotions.reject(&:expired?)
```

Comment:
@nishant-cyro nice catch  :+1: - In that case we can make the **possible_promotions** itself an association on Spree::Product instead of having the **advertised_promotions** as above. wdyt ?